docs/changelog/0.4.0.md

```diff
@@ -0,0 +1,532 @@
+
+## Native lib and unsafe package changes
+### CFuncPtr changes
+You no longer need to implement the `CFuncPtrN` trait which is now private for Scala Native implementation.
+Instead, you can use an implicit conversion method taking arbitrary `scala.FunctionN` and returning `CFuncPtrN`.
+
+```scala
+type Callback = CFuncPtr1[CInt,Unit]
+def registerCallback(cFn: Callback) = extern
+def fn(n: CInt): Unit = ???
+
+registerCallback(CFuncPtr1.fromScalaFunction(fn))
+registerCallback(fn)
+
+registerCallback{n: CInt => println("hello native")}
+```
+
+It's now also possible to work with an arbitrary pointer and convert it to `CFuncPtrN` that can be called in your Scala code
+or to convert your function to any pointer if your native library needs this.
+
+```scala
+import scala.scalanative.unsafe.Ptr
+val cFnPtr: CFuncPtr0[CInt] = ???
+val fnPtr: Ptr[Byte]    = CFuncPtr.toPtr(cFnPtr)
+val fnFromPtr:       = CFuncPtr.fromPtr[CFuncPtr0[CInt]](fnPtr)
+```
+
+### Unsigned CSize
+Since 0.4.0 version we are deprecation usage of ignored numeral types for methods expecting CSize which now is of type `ULong`.
+
+### Commix GC
+This release also adds a new Garbage Collector - Commix, a parallel mark, and concurrent sweep GC, based on the well known Immix GC.
+It reduces GC pause times by utilizing additional processor cores during mark and sweep phases. It was added soon after
+the 0.4.0-M2 release, but was only available via an unofficial snapshot.
+
+Remember, it **does not** mean Scala Native is starting to officially support multi-threading!
+Commix GC was written in C and uses `pthread` to work. In case if your application needs concurrency support you may try experimental library [scala-native-loop](https://github.com/scala-native/scala-native-loop)
+
+
+## Java standard library changes
+### Removal of java.time / java.text
+This release comes with removal of `java.time` and `java.text` packages for Scala Native core. This decision was taken
+in order to allow for proper implementation of those modules in third-party library. It's specially important for classes
+like `Locale` that are hard to maintain in Scala Native core repo.
+
+Removal of mentioned modules does not create regression in other definitions as long as they're not using directly or transitively one of removed classes.
+```scala
+val str: String = "Hello Native"
+str.toLowerCase()              // works as before
+str.toLowerCase(Locale.French) // fails while linking if not provided implementation
+```
+  
+
+### Unicode specification
+We have updated our `java.lang.String` and `java.lang.Character` to be compatible with Unicode 13.0.0 specification,
+allowing us to replace relatively old standard 6.3.0 used in JDK 8
+Additionally we've implemented missing handling of `java.lang.String.to{Lower, Upper}Case` methods for 
+special and language specific casing rules. Further one requires linking implementation for `Locale`. 
```

Comment:
There's a whole host a new methods and classes that have appeared in the javalib. Either we should exhaustively list them, or just mention them as a whole, but we shouldn't single out `to{Lower,Upper}Case`.

Reply:
I've replaced it with mention of adding multiple missing javalib definitions as part of bugfixes